src/plone/restapi/serializer/user.py

```diff
@@ -19,6 +22,19 @@ def __init__(self, context, request):
         self.context = context
         self.request = request
 
+    @property
+    def is_zope_manager(self):
+        return check_permission(ManagePortal, self.context)
+
+    @property
+    def can_manage_users(self):
+        return check_permission(PloneManageUsers, self.context)
+
+    def can_delete(self, roles):
+        if self.is_zope_manager:
+            return True
+        return "Manager" not in roles
+
```

Comment:
@ksuess today the `can_delete` key is only returned if the current user can manage user. See:

https://github.com/plone/plone.restapi/blob/08f8f871bbf63c4f1033cc3f2c46d371e994aa22/src/plone/restapi/serializer/user.py#L54-L55

I can test `self.can_manage_users` inside the `can_delete` method too, but it would be strange because I would have to test it twice. What is your opinion?

Reply:
Enhancing data with an additional key value pair for all cases is more reliable and better to handle by the consumer.

```
class BaseSerializer:
    def __init__(self, context, request):
        self.context = context
        self.request = request

    def _can_be_managed_by_current_user(self, roles):
        if check_permission(ManagePortal, self.context):
            return True
        # Allow Site Administrator to manage users that don't have Manager role.
        return (
            check_permission(PloneManageUsers, self.context) and "Manager" not in roles
        )

    def __call__(self):
        user = self.context
        portal = getSite()

        # Roles of context (to be serialized user)
        roles = user.getRoles()
        # Anonymous and Authenticated are pseudo roles assign automatically
        # to logged-in or logged-out users. They should not be exposed here.
        roles = sorted(list(set(roles) - {"Anonymous", "Authenticated"}))

        data = {
            "@id": f"{portal.absolute_url()}/@users/{user.id}",
            "id": user.id,
            "username": user.getUserName(),
            "roles": roles,
            "can_be_managed_by_current_user": self._can_be_managed_by_current_user(
                roles
            ),
        }
```

Reply:
If we have the authenticated user roles in the frontend, we can remove the can_assign and can_delete keys from the endpoints. I will remove them.

Reply:
@ksuess I removed the `can_delete key` from the users endpoint.